Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.h

```diff
@@ -25,9 +25,17 @@ namespace GradientSignal
         , private ImageGradientModificationNotificationBus::Handler
     {
     public:
+        AZ_RTTI(EditorImageGradientComponentMode, "{49957D52-F1C3-4C34-AA84-7661BC418AB2}", EditorBaseComponentMode)
+
         EditorImageGradientComponentMode(const AZ::EntityComponentIdPair& entityComponentIdPair, AZ::Uuid componentType);
         ~EditorImageGradientComponentMode() override;
 
+        static void Reflect(AZ::ReflectContext* context);
```

Comment:
Just curious, why is the reflection needed?

Reply:
The new Component Mode action handler gets a list of all component modes to iterate over from the reflection system, so we don't have to add registrations for them in a single place.